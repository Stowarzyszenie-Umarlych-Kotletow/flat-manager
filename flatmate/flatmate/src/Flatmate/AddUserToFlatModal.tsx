import {Modal, ModalContent, ModalTitle} from "react-native-modals";
import {Controller, useForm} from "react-hook-form";
import * as React from "react";
import {useState} from "react";
import styles from "../static/styles";
import {Text, TextInput} from "react-native";
import {Button} from "react-native-elements";

function parseData(data) {
    let parsedData = {
        "username": data.username,
    }
    return parsedData;
}

export function AddUserToFlatModal({showAddUserToFlatModal, setShowAddUserToFlatModal}) {

    const [showUsernameWarning, setUsernameWarning] = useState(false);

    const {control, handleSubmit, formState: {errors}} = useForm({
        defaultValues: {
            username: '',
        },
    });

    async function addUser(data) {
        if (data.username) {
            data = parseData(data);
            // send user to backend
            setUsernameWarning(false);
        } else {
            setUsernameWarning(true);
        }
    }


    return (<Modal
            width={0.9}
            rounded
            actionsBordered
            style={{zIndex: 1000}}
            visible={showAddUserToFlatModal}
            modalTitle={<ModalTitle title="Add user" align="left"/>}
            onTouchOutside={() => {
                setShowAddUserToFlatModal(false);
            }}
        >
            <ModalContent>
                <Controller
                    control={control}
                    rules={{
                        maxLength: 100,
                    }}
                    render={({field: {onChange, onBlur, value}}) => (<TextInput
                            style={styles.accFormTextInput}
                            onBlur={onBlur}
                            onChangeText={onChange}
                            value={value}
                            placeholder="Username"
                        />)}
                    name="username"
                />
                {!showUsernameWarning ? null : <Text style={styles.warningText}> Username cannot be empty </Text>}
                <Button
                    buttonStyle={styles.blueButton}
                    title="Add User"
                    onPress={handleSubmit(addUser)}
                />
                <Button
                    buttonStyle={styles.blueButton}
                    title="Close"
                    onPress={() => {
                        setShowAddUserToFlatModal(false);
                    }}
                />
            </ModalContent>
        </Modal>)
}