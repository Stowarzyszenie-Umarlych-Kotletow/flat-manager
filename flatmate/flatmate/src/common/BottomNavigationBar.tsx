import * as React from "react";
import {Text, TouchableOpacity} from "react-native";
import styles from "../static/styles";

const calendar = require("../static/calendar.svg") as string;
const users = require("../static/users.svg") as string;
const gear = require("../static/gear.svg") as string;
const bill = require("../static/bill.svg") as string;
const dashboard = require("../static/dashboard.svg") as string;
const tasks = require("../static/tasks.svg") as string;


export function BottomNavigationBar({openUsers, openTasks, openCalendar, openTransactionManager, openDashboard}) {
    return (
        <footer style={{
            backgroundColor: '#3333ff',
            display: "flex",
            paddingBlock: '5px',
            bottom: '0',
            left: '0',
            width: '100%',
            justifyContent: "space-around",
            alignItems: "center",
            minHeight: '75px',
        }}>
            <TouchableOpacity
                onPress={() => openTasks()}
                style={styles.columnView}
            >
                <img src={tasks} alt=" " style={{width: '35px', height: '35px'}}/>
                <Text> Tasks </Text>
            </TouchableOpacity>
            <TouchableOpacity
                onPress={() => {openTransactionManager()}}
                style={styles.columnView}
            >
                <img src={bill} alt=" " style={{width: '35px', height: '35px'}}/>
                <Text> Transactions </Text>
            </TouchableOpacity>
            <TouchableOpacity
                onPress={() => openDashboard()}
                style={styles.columnView}
            >
                <img src={dashboard} alt=" " style={{width: '35px', height: '35px'}}/>
                <Text> Home </Text>
            </TouchableOpacity>
            <TouchableOpacity
                onPress={() => openUsers()}
                style={styles.columnView}
            >
                <img src={users} alt=" " style={{width: '35px', height: '35px'}}/>
                <Text> Flatmates </Text>
            </TouchableOpacity>
            <TouchableOpacity
                onPress={() => openCalendar()}
                style={styles.columnView}
            >
                <img src={calendar} alt=" " style={{width: '35px', height: '35px'}}/>
                <Text> Calendar </Text>
            </TouchableOpacity>
            


            {/* <TouchableOpacity
                onPress={() => {console.log("flat settings")}}
                style={styles.columnView}
            >
                <img src={gear} alt="open settings" style={{width: '35px', height: '35px'}}/>
                <Text> Flat Settings</Text>
            </TouchableOpacity> */}

            
        </footer>
    )
}