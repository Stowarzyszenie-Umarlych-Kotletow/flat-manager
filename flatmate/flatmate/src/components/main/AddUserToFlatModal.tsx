import { Modal, ModalContent, ModalTitle } from "react-native-modals";
import { Controller, useForm } from "react-hook-form";
import * as React from "react";
import { useState } from "react";
import styles from "../../static/styles";
import { Text, TextInput, ScrollView } from "react-native";
import { Button } from "react-native-elements";
import { useAppDispatch} from "../../store";
import { useFlat } from "../../features/hooks";
import {useGetUserByUsernameQuery, userApi} from "../../features/api/user-api";
import { useAddUserToFlatMutation } from "../../features/api/flat-api";
import { UserRole } from "../../models/flat.model"
import { showMessage } from "react-native-flash-message";

export function AddUserToFlatModal({ setShowAddUserToFlatModal }) {

  const [usernameWarning, setUsernameWarning] = useState(null);
  const dispatch = useAppDispatch();
  const { flatId } = useFlat();
  const [addUser, {status}] = useAddUserToFlatMutation();

  const { control, handleSubmit, formState: { errors } } = useForm({
    defaultValues: {
      username: '',
    },
  });

  async function completeAddUser(data) {
    let warning: string = null;
    const username = data.username;
    if (username) {
      try {
        const action = userApi.endpoints.getUserByUsername.initiate({username});
        const userId = (await dispatch(action).unwrap()).id;
        const role = UserRole.USER;
        await addUser({ flatId, userId, role}).unwrap();

        setShowAddUserToFlatModal(false);
      } catch (err) {
        showMessage({
          message: `Cannot add user (error ${err.status})`,
          type: "danger"
        })
      }
    } else {
      showMessage({
        message: "Username cannot be empty",
        type: "danger"
      })
    }
    setUsernameWarning(warning);
  }


  return (
  <Modal
    width={0.9}
    rounded
    actionsBordered
    style={{ zIndex: 1000 }}
    visible={true}
    modalTitle={<ModalTitle title="Add user" align="left" />}
    onTouchOutside={() => {
      setShowAddUserToFlatModal(false);
    }}
  >
	<ScrollView>
	<ModalContent>
		<Controller
			control={control}
			rules={{
				maxLength: 100,
			}}
			render={({ field: { onChange, onBlur, value } }) => (<TextInput
				style={styles.textInput}
				onBlur={onBlur}
				onChangeText={onChange}
				value={value}
				placeholder="Username"
			/>)}
			name="username"
		/>
		{!usernameWarning == null ? null : <Text style={styles.warningText}>{usernameWarning}</Text>}
		<Button
			buttonStyle={styles.greenButton}
			title="Add User"
			onPress={handleSubmit(completeAddUser)}
		/>
		<Button
			buttonStyle={styles.orangeButton}
			title="Close"
			onPress={() => {
				setShowAddUserToFlatModal(false);
			}}
		/>
	</ModalContent>
	</ScrollView>
  </Modal>)
}