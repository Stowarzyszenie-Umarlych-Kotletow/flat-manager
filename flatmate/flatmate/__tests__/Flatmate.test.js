import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { mockStore } from "./utils/test-utils";
import { ModalPortal } from 'react-native-modals';
import { AddUserToFlatModal } from '../src/components/main/AddUserToFlatModal';
import { BottomNavigationBar } from '../src/components/main/BottomNavigationBar';
import { CreateFlatModal } from '../src/components/main/CreateFlatModal';
import { DashboardScreen } from '../src/screens/main/DashboardScreen';
import { ManageFlatsScreen } from '../src/screens/main/ManageFlatsScreen';




test('AddUserToFlatModal Screen renders correctly', () => {
    const tree =  renderer.create(<Provider store={mockStore()}><AddUserToFlatModal /> <ModalPortal/> </Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('BottomNavigationBar Screen renders correctly', () => {
    const tree =  renderer.create(<Provider store={mockStore()}><BottomNavigationBar /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('CreateFlatModal Screen renders correctly', () => {
    const tree =  renderer.create(<Provider store={mockStore()}><CreateFlatModal /> <ModalPortal/> </Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('DashboardScreen Screen renders correctly', () => {
    const tree =  renderer.create(<Provider store={mockStore()}><DashboardScreen /> <ModalPortal/> </Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('ManageFlatsScreen Screen renders correctly', () => {
    const tree =  renderer.create(<Provider store={mockStore()}><ManageFlatsScreen /> <ModalPortal/> </Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});


