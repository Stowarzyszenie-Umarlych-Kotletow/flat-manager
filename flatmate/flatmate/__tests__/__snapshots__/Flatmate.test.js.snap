// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUserToFlatModal Screen renders correctly 1`] = `
Array [
  " ",
  <View
    pointerEvents="auto"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        false,
      ]
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "zIndex": 1000,
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "backgroundColor": "#000",
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "left": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 8,
              "height": null,
              "opacity": 0,
              "overflow": "hidden",
              "width": 675,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                  "padding": 14,
                  "paddingHorizontal": 18,
                },
                Object {
                  "alignItems": "flex-start",
                },
                Object {
                  "backgroundColor": "#F9F9FB",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#DAD9DC",
                },
                undefined,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#151822",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "500",
                  },
                  undefined,
                ]
              }
            >
              Add user
            </Text>
          </View>
          <RCTScrollView>
            <View>
              <View
                style={
                  Array [
                    Object {
                      "paddingHorizontal": 18,
                      "paddingVertical": 24,
                    },
                    Object {
                      "paddingTop": 0,
                    },
                    undefined,
                  ]
                }
              >
                <TextInput
                  allowFontScaling={true}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="Username"
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "backgroundColor": "#fafafa",
                      "borderColor": "#eaeaea",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "fontSize": 14,
                      "height": 45,
                      "marginBottom": 5,
                      "marginLeft": 5,
                      "marginRight": 5,
                      "marginTop": 5,
                      "minWidth": 100,
                      "paddingLeft": 10,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
                <Text
                  style={
                    Object {
                      "color": "red",
                      "fontSize": 13,
                      "fontWeight": "bold",
                      "marginBottom": 5,
                      "marginTop": 5,
                      "textAlign": "center",
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 3,
                        "overflow": "hidden",
                      },
                      Object {
                        "borderRadius": 3,
                      },
                      undefined,
                      false,
                    ]
                  }
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      Object {
                        "busy": false,
                        "disabled": false,
                      }
                    }
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#5cd07f",
                          "borderColor": "#2089dc",
                          "borderRadius": 5,
                          "borderWidth": 0,
                          "flexDirection": "row",
                          "height": 45,
                          "justifyContent": "center",
                          "marginLeft": 25,
                          "marginRight": 25,
                          "marginTop": 10,
                          "padding": 8,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "white",
                            "fontSize": 18,
                            "paddingVertical": 1,
                            "textAlign": "center",
                          }
                        }
                      >
                        Add User
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 3,
                        "overflow": "hidden",
                      },
                      Object {
                        "borderRadius": 3,
                      },
                      undefined,
                      false,
                    ]
                  }
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      Object {
                        "busy": false,
                        "disabled": false,
                      }
                    }
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#ff9933",
                          "borderColor": "#2089dc",
                          "borderRadius": 5,
                          "borderWidth": 0,
                          "flexDirection": "row",
                          "height": 45,
                          "justifyContent": "center",
                          "marginLeft": 25,
                          "marginRight": 25,
                          "marginTop": 10,
                          "padding": 8,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "white",
                            "fontSize": 18,
                            "paddingVertical": 1,
                            "textAlign": "center",
                          }
                        }
                      >
                        Close
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </View>,
  " ",
]
`;

exports[`BottomNavigationBar Screen renders correctly 1`] = `
<footer
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#009999",
      "bottom": "0",
      "display": "flex",
      "height": "75px",
      "justifyContent": "space-evenly",
      "left": "0",
      "paddingInline": "5px",
      "position": "fixed",
      "width": "100%",
      "zIndex": 100,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "opacity": 1,
      }
    }
  >
    <img
      alt=" "
      src={Object {}}
      style={
        Object {
          "height": "35px",
          "width": "35px",
        }
      }
    />
    <Text>
       Calendar 
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "opacity": 1,
      }
    }
  >
    <img
      alt=" "
      src={Object {}}
      style={
        Object {
          "height": "35px",
          "width": "35px",
        }
      }
    />
    <Text>
       Tasks 
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "opacity": 1,
      }
    }
  >
    <img
      alt=" "
      src={Object {}}
      style={
        Object {
          "height": "35px",
          "width": "35px",
        }
      }
    />
    <Text>
       Home 
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "opacity": 1,
      }
    }
  >
    <img
      alt=" "
      src={Object {}}
      style={
        Object {
          "height": "35px",
          "width": "35px",
        }
      }
    />
    <Text>
       Mates 
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "opacity": 1,
      }
    }
  >
    <img
      alt=" "
      src={Object {}}
      style={
        Object {
          "height": "35px",
          "width": "35px",
        }
      }
    />
    <Text>
       Bills 
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "opacity": 1,
      }
    }
  >
    <img
      alt=" "
      src={Object {}}
      style={
        Object {
          "height": "35px",
          "width": "35px",
        }
      }
    />
    <Text>
       Debts 
    </Text>
  </View>
</footer>
`;

exports[`CreateFlatModal Screen renders correctly 1`] = `
Array [
  " ",
  <View
    pointerEvents="auto"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        false,
      ]
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "zIndex": 1000,
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "backgroundColor": "#000",
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "left": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 8,
              "height": null,
              "opacity": 0,
              "overflow": "hidden",
              "width": 675,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                  "padding": 14,
                  "paddingHorizontal": 18,
                },
                Object {
                  "alignItems": "flex-start",
                },
                Object {
                  "backgroundColor": "#F9F9FB",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#DAD9DC",
                },
                undefined,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#151822",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "500",
                  },
                  undefined,
                ]
              }
            >
              Create flat
            </Text>
          </View>
          <RCTScrollView>
            <View>
              <View
                style={
                  Array [
                    Object {
                      "paddingHorizontal": 18,
                      "paddingVertical": 24,
                    },
                    Object {
                      "paddingTop": 0,
                    },
                    undefined,
                  ]
                }
              >
                <TextInput
                  allowFontScaling={true}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="Name"
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "backgroundColor": "#fafafa",
                      "borderColor": "#eaeaea",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "fontSize": 14,
                      "height": 45,
                      "marginBottom": 5,
                      "marginLeft": 5,
                      "marginRight": 5,
                      "marginTop": 5,
                      "minWidth": 100,
                      "paddingLeft": 10,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 3,
                        "overflow": "hidden",
                      },
                      Object {
                        "borderRadius": 3,
                      },
                      undefined,
                      false,
                    ]
                  }
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      Object {
                        "busy": false,
                        "disabled": false,
                      }
                    }
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#5cd07f",
                          "borderColor": "#2089dc",
                          "borderRadius": 5,
                          "borderWidth": 0,
                          "flexDirection": "row",
                          "height": 45,
                          "justifyContent": "center",
                          "marginLeft": 25,
                          "marginRight": 25,
                          "marginTop": 10,
                          "padding": 8,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "white",
                            "fontSize": 18,
                            "paddingVertical": 1,
                            "textAlign": "center",
                          }
                        }
                      >
                        Submit
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 3,
                        "overflow": "hidden",
                      },
                      Object {
                        "borderRadius": 3,
                      },
                      undefined,
                      false,
                    ]
                  }
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      Object {
                        "busy": false,
                        "disabled": false,
                      }
                    }
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#ff9933",
                          "borderColor": "#2089dc",
                          "borderRadius": 5,
                          "borderWidth": 0,
                          "flexDirection": "row",
                          "height": 45,
                          "justifyContent": "center",
                          "marginLeft": 25,
                          "marginRight": 25,
                          "marginTop": 10,
                          "padding": 8,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "white",
                            "fontSize": 18,
                            "paddingVertical": 1,
                            "textAlign": "center",
                          }
                        }
                      >
                        Cancel
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </View>,
  " ",
]
`;

exports[`DashboardScreen Screen renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": 1259,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "800",
          "marginBottom": 15,
          "marginTop": 15,
          "textAlign": "center",
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "500",
          "marginBottom": 20,
          "marginTop": 20,
          "textAlign": "center",
        }
      }
    >
      No debts! 😎
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "500",
          "marginBottom": 20,
          "marginTop": 20,
          "textAlign": "center",
        }
      }
    >
      No tasks for today! 🎉
    </Text>
    <RCTScrollView
      style={
        Object {
          "paddingBottom": 95,
          "paddingHorizontal": 10,
        }
      }
    >
      <View />
    </RCTScrollView>
    <footer
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#009999",
          "bottom": "0",
          "display": "flex",
          "height": "75px",
          "justifyContent": "space-evenly",
          "left": "0",
          "paddingInline": "5px",
          "position": "fixed",
          "width": "100%",
          "zIndex": 100,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "opacity": 1,
          }
        }
      >
        <img
          alt=" "
          src={Object {}}
          style={
            Object {
              "height": "35px",
              "width": "35px",
            }
          }
        />
        <Text>
           Calendar 
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "opacity": 1,
          }
        }
      >
        <img
          alt=" "
          src={Object {}}
          style={
            Object {
              "height": "35px",
              "width": "35px",
            }
          }
        />
        <Text>
           Tasks 
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "opacity": 1,
          }
        }
      >
        <img
          alt=" "
          src={Object {}}
          style={
            Object {
              "height": "35px",
              "width": "35px",
            }
          }
        />
        <Text>
           Home 
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "opacity": 1,
          }
        }
      >
        <img
          alt=" "
          src={Object {}}
          style={
            Object {
              "height": "35px",
              "width": "35px",
            }
          }
        />
        <Text>
           Mates 
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "opacity": 1,
          }
        }
      >
        <img
          alt=" "
          src={Object {}}
          style={
            Object {
              "height": "35px",
              "width": "35px",
            }
          }
        />
        <Text>
           Bills 
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "opacity": 1,
          }
        }
      >
        <img
          alt=" "
          src={Object {}}
          style={
            Object {
              "height": "35px",
              "width": "35px",
            }
          }
        />
        <Text>
           Debts 
        </Text>
      </View>
    </footer>
  </View>,
  " ",
  " ",
]
`;

exports[`ManageFlatsScreen Screen renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": "calc(100vh - 75px)",
        "overflow": "scroll",
        "paddingBottom": "20px",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "800",
          "marginBottom": 15,
          "marginTop": 15,
          "textAlign": "center",
        }
      }
    >
      Manage flats
    </Text>
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 3,
          },
          undefined,
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#5cd07f",
              "borderColor": "#2089dc",
              "borderRadius": 5,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "marginLeft": 25,
              "marginRight": 25,
              "marginTop": 10,
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Create flat
          </Text>
        </View>
      </View>
    </View>
  </View>,
  " ",
  " ",
]
`;
